tests/pytests/test_vecsim.py

```diff
@@ -1073,3 +1069,121 @@ def test_hybrid_query_change_policy():
                          'tag1', str(1), 'tag2', 'word'+str(file))
     res = execute_hybrid_query(env, query_string, vectors[0], 'tag2', hybrid_mode='HYBRID_BATCHES_TO_ADHOC_BF').res
     env.assertEqual(res[:2], [1, str(n)])
+
+
+def test_memory_limits():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+
+    system_memory = int(env.cmd('info', 'memory')['total_system_memory'])
+    used_memory = int(env.cmd('info', 'memory')['used_memory'])
+    currIdx = 0
+
+    # OK parameters
+    env.assertOk(conn.execute_command('FT.CREATE', currIdx, 'SCHEMA', 'v', 'VECTOR', 'FLAT', '10', 'TYPE', 'FLOAT32',
+                                      'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', 10000, 'BLOCK_SIZE', 100))
+    currIdx+=1
+
+    env.assertOk(conn.execute_command('FT.CREATE', currIdx, 'SCHEMA', 'v', 'VECTOR', 'HNSW', '8', 'TYPE', 'FLOAT32',
+                                      'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', 10000))
+    currIdx+=1
+
+    # Index initial size exceeded limits
+    env.expect('FT.CREATE', currIdx, 'SCHEMA', 'v', 'VECTOR', 'HNSW', '8', 'TYPE', 'FLOAT32',
+               'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', system_memory).error().contains(
+               'Vector index size exceeded server limit')
+    currIdx+=1
+
+    env.expect('FT.CREATE', currIdx, 'SCHEMA', 'v', 'VECTOR', 'FLAT', '10', 'TYPE', 'FLOAT32',
+               'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', system_memory, 'BLOCK_SIZE', 100).error().contains(
+               'Vector index size exceeded server limit')
+    currIdx+=1
+    env.expect('FT.CREATE', currIdx, 'SCHEMA', 'v', 'VECTOR', 'FLAT', '10', 'TYPE', 'FLOAT32',
+               'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', 100, 'BLOCK_SIZE', system_memory).error().contains(
+               'Vector index size exceeded server limit')
+    currIdx+=1
+
+    # Block size with no limits fail
+    env.expect('FT.CREATE', currIdx, 'SCHEMA', 'v', 'VECTOR', 'FLAT', '10', 'TYPE', 'FLOAT32',
+               'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', 100, 'BLOCK_SIZE', system_memory // 100).error().contains(
+               'Vector index block size exceeded server limit')
+    currIdx+=1
+
+    # TODO: uncomment when BLOCK_SIZE is added to FT.CREATE on HNSW
+    # env.expect('FT.CREATE', currIdx, 'SCHEMA', 'v', 'VECTOR', 'HNSW', '10', 'TYPE', 'FLOAT32',
+    #            'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', 100, 'BLOCK_SIZE', system_memory // 100).error().contains(
+    #            'Vector index block size exceeded server limit')
+    # currIdx+=1
+
+    # Config max memory (redis server memory limit)
+    maxmemory = used_memory * 10
```

Comment:
you are using this metric, where it is given in bytes, as a vector number parameter, seems a bit awkward IMO. What is the motivation for those tests if you translate the memory amount into vector amount a few tests after?

Reply:
with initial capacity set to the memory limit in bytes, I know that the memory limit will surely be exceeded.